pkg/tsdb/legacydata/time_range.go

```diff
@@ -21,99 +21,167 @@ func NewDataTimeRange(from, to string) DataTimeRange {
 	}
 }
 
-func (tr *DataTimeRange) GetFromAsMsEpoch() int64 {
+func (tr DataTimeRange) GetFromAsMsEpoch() int64 {
 	return tr.MustGetFrom().UnixNano() / int64(time.Millisecond)
 }
 
-func (tr *DataTimeRange) GetFromAsSecondsEpoch() int64 {
+func (tr DataTimeRange) GetFromAsSecondsEpoch() int64 {
 	return tr.GetFromAsMsEpoch() / 1000
 }
 
-func (tr *DataTimeRange) GetFromAsTimeUTC() time.Time {
+func (tr DataTimeRange) GetFromAsTimeUTC() time.Time {
 	return tr.MustGetFrom().UTC()
 }
 
-func (tr *DataTimeRange) GetToAsMsEpoch() int64 {
+func (tr DataTimeRange) GetToAsMsEpoch() int64 {
 	return tr.MustGetTo().UnixNano() / int64(time.Millisecond)
 }
 
-func (tr *DataTimeRange) GetToAsSecondsEpoch() int64 {
+func (tr DataTimeRange) GetToAsSecondsEpoch() int64 {
 	return tr.GetToAsMsEpoch() / 1000
 }
 
-func (tr *DataTimeRange) GetToAsTimeUTC() time.Time {
+func (tr DataTimeRange) GetToAsTimeUTC() time.Time {
 	return tr.MustGetTo().UTC()
 }
 
-func (tr *DataTimeRange) MustGetFrom() time.Time {
+func (tr DataTimeRange) MustGetFrom() time.Time {
 	res, err := tr.ParseFrom()
 	if err != nil {
 		return time.Unix(0, 0)
 	}
 	return res
 }
 
-func (tr *DataTimeRange) MustGetTo() time.Time {
+func (tr DataTimeRange) MustGetTo() time.Time {
 	res, err := tr.ParseTo()
 	if err != nil {
 		return time.Unix(0, 0)
 	}
 	return res
 }
 
-func (tr DataTimeRange) ParseFrom() (time.Time, error) {
-	return parseTimeRange(tr.From, tr.Now, false, nil)
+func (tr DataTimeRange) ParseFrom(options ...TimeRangeOption) (time.Time, error) {
+	options = append(options, WithNow(tr.Now))
+
+	pt := newParsableTime(tr.From, options...)
+	return pt.Parse()
 }
 
-func (tr DataTimeRange) ParseTo() (time.Time, error) {
-	return parseTimeRange(tr.To, tr.Now, true, nil)
+func (tr DataTimeRange) ParseTo(options ...TimeRangeOption) (time.Time, error) {
+	options = append(options, WithRoundUp(), WithNow(tr.Now))
+
+	pt := newParsableTime(tr.To, options...)
+	return pt.Parse()
 }
 
-func (tr DataTimeRange) ParseFromWithLocation(location *time.Location) (time.Time, error) {
-	return parseTimeRange(tr.From, tr.Now, false, location)
+func (tr DataTimeRange) ParseFromWithWeekStart(location *time.Location, weekstart time.Weekday) (time.Time, error) {
+	return tr.ParseFrom(WithLocation(location), WithWeekstart(weekstart))
 }
 
-func (tr DataTimeRange) ParseToWithLocation(location *time.Location) (time.Time, error) {
-	return parseTimeRange(tr.To, tr.Now, true, location)
+func (tr DataTimeRange) ParseToWithWeekStart(location *time.Location, weekstart time.Weekday) (time.Time, error) {
+	return tr.ParseTo(WithLocation(location), WithWeekstart(weekstart))
 }
```

Comment:
Without keeping these methods, we'd break Grafana Enterprise on merge. They can be removed once Grafana Enterprise has been updated to use the new functional options.